app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  def update_posts_likes_counter
+    author.increment!(:posts_count)
+  end
+
+  def most_recent_comments
+    comments.order(created_at: :desc).limit(5)
+  end
+end
```

Comment:
It will be nice to use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_post_counter method every time a new post has been saved to the database, then make the update_post_counter method private.
```ruby
# app/models/post.rb
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```